src/Castle.Core/DynamicProxy/Contributors/ClassProxySerializableContributor.cs

```diff
@@ -187,7 +191,11 @@ private bool VerifyIfBaseImplementsGetObjectData(Type baseType, IList<MethodInfo
 				throw new ArgumentException(message);
 			}
 
-			methodsToSkip.Add(getObjectDataMethod);
+			if (model.Methods.FirstOrDefault(m => m.Method == getObjectDataMethod) is { } method &&
+				method.Proxyable)
+			{
+				method.Ignore = true;
+			}
```

Comment:
This was the _only_ place where anything ever got added to `methodsToSkip`, btw.